### PR TITLE
Add gcflags to debug builds

### PR DESCRIPTION
**Pull request by bryanhuhta**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -116,30 +117,30 @@ go/deps:
...
 
 .PHONY: go/bin-pyroscope-debug
 go/bin-pyroscope-debug:
-	$(call go_build_pyroscope,$(GO_LDFLAGS))
+	$(call go_build_pyroscope,$(GO_LDFLAGS),$(GO_GCFLAGS))
 
 .PHONY: go/bin-profilecli-debug
 go/bin-profilecli:
-	$(call go_build_profilecli,$(GO_LDFLAGS))
+	$(call go_build_profilecli,$(GO_LDFLAGS),$(GO_GCFLAGS))
```
Comment (korniltsev) [inline Makefile:143]: not related to your changes, something seems wrong here - phony directive and the target name is different

```
.PHONY: go/bin-profilecli-debug
go/bin-profilecli:
```

File: Makefile [hunk 2]
```diff
@@ -30,6 +30,7 @@ EMBEDASSETS ?= embedassets
 # Build flags
 VPREFIX := github.com/grafana/pyroscope/pkg/util/build
 GO_LDFLAGS   := -X $(VPREFIX).Branch=$(GIT_BRANCH) -X $(VPREFIX).Version=$(IMAGE_TAG) -X $(VPREFIX).Revision=$(GIT_REVISION) -X $(VPREFIX).BuildDate=$(GIT_LAST_COMMIT_DATE)
+GO_GCFLAGS   := all="-N -l"
```
Comment (korniltsev) [inline Makefile:33]: should we add ad DEBUG preffix or suffix here ?

**Review by korniltsev — APPROVED** [review]

File: Makefile [hunk 3]
```diff
@@ -116,30 +117,30 @@ go/deps:
...
-	$(call go_build_profilecli,-s -w $(GO_LDFLAGS))
+	$(call go_build_pyroscope,-s -w $(GO_LDFLAGS),)
+	$(call go_build_profilecli,-s -w $(GO_LDFLAGS),)
 
 .PHONY: go/bin-pyroscope-debug
 go/bin-pyroscope-debug:
-	$(call go_build_pyroscope,$(GO_LDFLAGS))
+	$(call go_build_pyroscope,$(GO_LDFLAGS),$(GO_GCFLAGS))
 
 .PHONY: go/bin-profilecli-debug
```
Comment (aleks-p) [inline Makefile:141]: nitpick and unrelated to this PR, this should probably match the make target (`go/bin-profilecli`)

**Review by aleks-p — APPROVED** [review]
LGTM

File: Makefile (hunk 2 above)
Comment (bryanhuhta) [reply Makefile]: Good idea, added in https://github.com/grafana/pyroscope/pull/3517/commits/8d1035ab702ad26a754b45863a736a00456973e8

File: Makefile (hunk 1 above)
Comment (bryanhuhta) [reply Makefile]: Haha nice catch, fixed in https://github.com/grafana/pyroscope/pull/3517/commits/8d1035ab702ad26a754b45863a736a00456973e8

File: Makefile (hunk 3 above)
Comment (bryanhuhta) [reply Makefile:141]: Fixed in https://github.com/grafana/pyroscope/pull/3517/commits/8d1035ab702ad26a754b45863a736a00456973e8!

**Pull request merged**